src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -370,15 +380,48 @@ private static bool TryParseEnumCore(ref Utf8JsonReader reader, JsonSerializerOp
             return success;
         }
 #else
-        private static bool TryParseEnumCore(string? enumString, JsonSerializerOptions options, out T value)
+        private bool TryParseEnumCore(string? enumString, JsonSerializerOptions options, out T value)
         {
+            if (!_converterOptions.HasFlag(EnumConverterOptions.AllowNumbers))
```

Comment:
`HasFlag` is not efficient in down-level runtimes (and it is not guaranteed to be always efficient in the current .NET runtime either).

This should use a regular bit check instead.

Reply:
Fixed.